src/umpire/ResourceManager.cpp

```diff
@@ -203,6 +227,21 @@ ResourceManager::getAllocator(int id)
   return Allocator(m_allocators_by_id[id]);
 }
 
+Allocator
+ResourceManager::getAllocatorFor(const resource::MemoryResourceTraits traits)
+{
+  UMPIRE_LOG(Debug, "(Looking up allocator by traits)");
+
+  for (auto r : m_resource_list) {
```

Comment:
I'd like to find a way to tie this to the "matcher" function of the resource factories, but for now, can you update it to take into account the default values of the traits?